ntbs-integration-tests/NotificationPages/ClinicalDetailsPageTests.cs

```diff
@@ -87,11 +91,57 @@ public async Task PostNotified_ReturnsPageWithAllModelErrors_IfModelNotValid()
             var resultDocument = await GetDocumentAsync(result);
 
             result.EnsureSuccessStatusCode();
-            resultDocument.AssertErrorMessage("other-site", "Site description is a mandatory field");
+            resultDocument.AssertErrorSummaryMessage("OtherSite-SiteDescription", "other-site", "Site name is a mandatory field");
             resultDocument.AssertErrorSummaryMessage("ClinicalDetails-BCGVaccinationYear", "bcg-vaccination", "BCG vaccination year is a mandatory field");
             resultDocument.AssertErrorSummaryMessage("ClinicalDetails-DeathDate", "postmortem", "Date of death is a mandatory field");
         }
 
+        [Fact]
+        public async Task ExpandableDiseaseSites_SingleOpen_IfValidModel()
+        {
+            // Arrange
+            var url = GetCurrentPathForId(Utilities.DRAFT_ID);
+
+            // Act
+            var document = await GetDocumentForUrl(url);
+
+            // Assert
+            var siteExpanders = document.QuerySelectorAll(".disease-sites");
+            Assert.Equal(2, siteExpanders.Length);
+            var openSiteExpanders = document.QuerySelectorAll(".disease-sites[open]");
+            Assert.Equal(1, openSiteExpanders.Length);
+        }
+
+        [Fact]
+        public async Task ExpandableDiseaseSites_BothOpen_IfSitesValidationError()
+        {
+            // Arrange
+            var url = GetCurrentPathForId(Utilities.NOTIFIED_ID);
+            var initialDocument = await GetDocumentForUrl(url);
+
+            var formData = new Dictionary<string, string>
+            {
+                ["NotificationId"] = Utilities.NOTIFIED_ID.ToString(),
+                ["NotificationSiteMap[OTHER]"] = "true",
+                ["OtherSite.SiteId"] = ((int)SiteId.OTHER).ToString(),
+                ["OtherSite.SiteDescription"] = null
+            };
+
+            // Act
+            var result = await SendPostFormWithData(initialDocument, formData, url, submitType:"Save");
+
+            // Assert
+            result.EnsureSuccessStatusCode();
+            var resultDocument = await GetDocumentAsync(result);
+
+            resultDocument.AssertErrorSummaryMessage("OtherSite-SiteDescription", "other-site", "Site name is a mandatory field");
+
+            var siteExpanders = resultDocument.QuerySelectorAll(".disease-sites");
+            Assert.Equal(2, siteExpanders.Length);
+            var openSiteExpanders = resultDocument.QuerySelectorAll(".disease-sites[open]");
+            Assert.Equal(2, openSiteExpanders.Length);

```

Comment:
These tests are a little clumsy, but thought they were worth it.

Reply:
Nice stuff!